### PR TITLE
Align Version and Worker Chips

### LINKED ISSUE #1694: Use v-chip for version chip
Currently, the version chip is actually a button. To align this with the other components on the shoot list, we should also use a v-chip.
Also, check if we can have a fixed width or fixed column width.

### PR DESCRIPTION
**Pull request by grolu**
**What this PR does / why we need it**:

Enhanced Version Chip
**Popper Information Display**: The version chip now features a popper that provides detailed information when clicked. This includes classification, expiration dates, and any deprecation warnings to keep you informed about the status of your version.
Screenshot 2024-03-21 at 16 41 49
Screenshot 2024-03-15 at 20 39 49

Worker Group Chips with Deprecation Warning
**Warning Color Indicator**: Worker group chips will now display in a warning color if they are marked as deprecated. This visual cue helps you quickly identify worker groups that need attention or updating and aligns the behavior with the version chip.
Screenshot 2024-03-16 at 15 02 08
Screenshot 2024-03-16 at 15 01 39

Streamlined Cluster Updates
**Updated Cluster Management**: The option to update a cluster has been moved to the actions menu on the cluster list page. This change makes it easier to access and perform cluster updates, as it is no longer accessible via the version chip. In addition, deprecation information is now also included in the version hints.
Screenshot 2024-03-16 at 15 00 23
Screenshot 2024-03-16 at 15 55 59<br />

Consistent Chip Display on Cluster Details
**Unified Chip Presentation**: To maintain consistency and ease of use, the version is now also presented as a chip on the cluster details page. This aligns with the presentation of worker chips, making the interface more intuitive and information more accessible.
Screenshot 2024-03-20 at 15 36 07

**Which issue(s) this PR fixes**:
Fixes #1694

**Special notes for your reviewer**:

**Release note**:

```feature user
Several improvements to version and worker chips:
- Version & Worker Group Chips: Now, both show in a warning color if deprecated, offering a clear visual cue for items needing updates. The version chip also reveals detailed info, including classification and expiration, with a click.
- Cluster Management: Updating clusters is now streamlined under the actions menu on the cluster list page for easier access.
- Consistent UI: Version information now appears as a chip on cluster details, aligning with the display of worker chips for a unified look.
```

### CODE REVIEW

**Comment by petersutter** [comment]
make dedicated row for classification

**Comment by grolu** [comment]
>make dedicated row for classification

done

File: frontend/src/components/ShootVersion/GShootVersionChip.vue [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    chipColor () {
+      return this.kubernetesVersion.isDeprecated ? 'warning' : 'primary'
+    },
+    kubernetesVersion () {
+      const version = find(this.kubernetesVersions(this.shootCloudProfileName), { version: this.shootK8sVersion })
+      if (!version) {
+        return {}
+      }
+      return version
+    },
```
Comment (holgerkoser) [inline frontend/src/components/ShootVersion/GShootVersionChip.vue:160]: Move this to shootItem and reuse in `GShootVersionConfiguration.vue`
```js
    ...
    supportedPatchAvailable () {
      return some(this.availableKubernetesUpdates?.patch, 'isSupported')
    },
    supportedUpgradeAvailable () {
      return some(this.availableKubernetesUpdates?.minor, 'isSupported')
    },
    availableKubernetesUpdates () {
      return this.availableKubernetesUpdatesForShoot(this.shootK8sVersion, this.shootCloudProfileName)
    },
    shootKubernetesVersionObject () {
      const kubernetesVersionObjects = this.kubernetesVersions(this.shootCloudProfileName)
      return find(kubernetesVersionObjects, ['version', this.shootK8sVersion]) ?? {}
    },
    ...
```

File: frontend/src/components/ShootVersion/GShootVersionConfiguration.vue [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        })
+      } catch (err) {
+        const errorMessage = 'Update Kubernetes version failed'
+        const errorDetails = errorDetailsFromError(err)
+        const detailedErrorMessage = errorDetails.detailedMessage
+        this.$refs.actionDialog.setError({ errorMessage, detailedErrorMessage })
+        this.logger.error(errorMessage, errorDetails.errorCode, errorDetails.detailedMessage, err)
+      }
+    },
+    async reset () {
```
Comment (holgerkoser) [inline frontend/src/components/ShootVersion/GShootVersionConfiguration.vue:178]:
```suggestion
    reset () {
```

File: frontend/src/components/ShootWorkers/GWorkerGroup.vue [hunk 3]
```diff
@@ -441,14 +451,11 @@ export default {
     machineImage () {
       const machineImages = this.machineImagesByCloudProfileName(this.cloudProfileName)
       const { name, version } = get(this.workerGroup, 'machine.image', {})
-      return find(machineImages, { name, version })
+      return find(machineImages, { name, version }) ?? {}
```
Comment (holgerkoser) [inline frontend/src/components/ShootWorkers/GWorkerGroup.vue:454]:
```suggestion
      return find(machineImages, ['name', version]) ?? {}
```

File: frontend/src/components/ShootVersion/GShootVersionConfiguration.vue [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+  },
+  computed: {
+    kubernetesVersion () {
+      const version = find(this.kubernetesVersions(this.shootCloudProfileName), { version: this.shootK8sVersion })
+      if (!version) {
+        return {}
+      }
+      return version
+    },
+    supportedPatchAvailable () {
```
Comment (holgerkoser) [inline frontend/src/components/ShootVersion/GShootVersionConfiguration.vue:118]: will be part of the `shootItem` mixin

File: frontend/src/components/ShootWorkers/GWorkerGroup.vue (hunk 3 above)
Comment (grolu) [reply frontend/src/components/ShootWorkers/GWorkerGroup.vue:454]: need to find image by name and version

**Review by petersutter — APPROVED** [review]
/lgtm

**Review by holgerkoser — APPROVED** [review]
/lgtm

**Pull request merged**